packages/web-app-files/src/components/SideBar/Details/FileDetails.vue

```diff
@@ -0,0 +1,213 @@
+<template>
+  <div id="oc-file-details-sidebar">
+    <oc-loader v-if="loading" />
+    <div v-else>
+      <div
+        v-if="highlightedFile.preview"
+        class="details-preview uk-flex uk-flex-middle uk-flex-center oc-mb-m"
+      >
+        <oc-img :src="highlightedFile.preview" alt="" />
+      </div>
+      <div
+        v-if="hasAnyShares && !isPublicPage"
```

Comment:
I'm a big fan of not having too much logic in the template. Would be nice to have an additional `showSharingInfo` computed property which has an early return `if (isPublicPage) { return false }` and then checks for shares.